src/renderer/components/BalanceGraph/BalanceGraph.tsx

```diff
@@ -0,0 +1,158 @@
+
+    const formatDate = (date: string): string=>{
+        const month = date.slice(4,7).toUpperCase();
+        const day = date.slice(8,10);
+        return month + " " + day;
+    };
+
+    const setXAxis = (array: Promise<number[]>, time: IntervalEnum): void=>{
+        const dataArray: Array<object> = [];
+        const dateToday = new Date();
+        switch (time) {
+            case IntervalEnum.HOUR: {
+                const minute = dateToday.getMinutes();
+                for (let i = 5; i >=0; i--) {
+                    dataArray.push({
+                        name: ((minute-10*i)<0 ? 60+(minute-10*i) : (minute-10*i)) + "'",
+                        value: array[i]
+                    });
+                }
+                break;
+            }
+            case IntervalEnum.DAY: {
+                const hour = dateToday.getHours();
+                for (let i = 9; i >=0; i--) {
+                    dataArray.push({
+                        name: (hour-i)<1 ? 24+hour-i : hour-i + "h",
+                        value: array[i]
+                    });
+                }
+                break;
+            }
+            case IntervalEnum.WEEK: {
+                const day = dateToday.getDay();
+                for (let i = 6; i >=0; i--) {
+                    dataArray.push({
+                        name: (day-i)<0 ? Days[7+day-i] : Days[day-i],
+                        value: array[i]
+                    });
+                }
+                break;
+            }
+            case IntervalEnum.MONTH: {
+                for (let i = 29; i >=0; i--) {
+                    const yesterday = new Date(new Date().setDate(new Date().getDate()-i-30));
+                    dataArray.push({
+                        name: formatDate(yesterday.toString()),
+                        value: array[i]
+                    });
+                }
+                break;
+            }
+            case IntervalEnum.YEAR: {
+                const month = dateToday.getMonth();
+                for (let i = 11; i >=0; i--) {
+                    dataArray.push({
+                        name: (month-i)<0 ? Months[12+month-i] : Months[month-i],
+                        value: array[i]
+                    });
+                }
+                break;
+            }
+        }
+        setData(dataArray);
+    };
+
+    useEffect(()=>{
+        const dataValueArray = props.getData();
+        switch (interval) {
+            case IntervalEnum.HOUR: setXAxis(dataValueArray, IntervalEnum.HOUR);
+                break;
+            case IntervalEnum.DAY: setXAxis(dataValueArray, IntervalEnum.DAY);
+                break;
+            case IntervalEnum.WEEK: setXAxis(dataValueArray, IntervalEnum.WEEK);
+                break;
+            case IntervalEnum.MONTH: setXAxis(dataValueArray, IntervalEnum.MONTH);
+                break;
+            case IntervalEnum.YEAR: setXAxis(dataValueArray, IntervalEnum.YEAR);
+                break;
```

Comment:
do you even need thos switch case. You could just do setXAxis(dataValueArray, interval)